cooldowns/cooldown.py

```diff
@@ -438,3 +439,227 @@ def bucket(self) -> CooldownBucketProtocol:
+        would return False if you invoked the Callable.
+    
+    cooldown_id: Optional[Union[int, str]]
+        Useful for resetting individual stacked cooldowns.
+        This should be unique globally,
+        behaviour is not guaranteed if not unique.
+
+        .. note::
+
+            This check will be given the same arguments as
+            the item you are applying the cooldown to.
+    
+    Usage
+    -----
+        - First create an instance of TriggerCooldown() with
+        the desired parameters.
+
+        ```
+        trigger_cooldown = cooldowns.TriggerCooldown(1, 5, cooldowns.SlashBucket.author)
+        ```
+        
+        - Then add the instance as a decorator to your command!
+
+        ```
+        @nextcord.slash_command()
+        @trigger_cooldown
+        async def command():
+        ```
+
+            The instance has to be defined in the same scope as the decorator!
+            Now, `command()` has applied a normal cooldown of `1 limit` and
+            `5 time_period`, as we defined it.
+
+        - Finally, inside your command, you can `trigger` the trigger cooldown:
+
+        ```
+        async def command():
+            # Do things
+            trigger_cooldown.trigger(30)
+            # You can still do things after this.
+            # Even you can `interaction.send()`.
+        ```
+
+            From the moment when the cooldown was triggered by `.trigger(30)`, every
+            single call to this command within 30 seconds will raise CallableOnCooldown!
+    
+    Raises
+    ------
+    `RuntimeError`
+        Expected the decorated function to be a coroutine.
+    `CallableOnCooldown`
+        This call resulted in a cooldown being put into effect.
+
+
+
+    """
+    def __init__(
+            self,
+            limit: int,
+            time_period: Union[float, datetime.timedelta],
+            bucket: Optional[CooldownBucketProtocol] = None,
+            *,
+            cooldown_id: Optional[Union[int, str]] = None,
+            check: Optional[MaybeCoro] = default_check,
+    ):
+        
+        self.triggered = False
+
+        self.limit = limit
+        self.time_period = time_period
+        self.bucket = bucket
+        self.cooldown_id = cooldown_id
+        self.check = check
+
+        # Normal Cooldown
+        self.cooldown = Cooldown(
+                limit= self.limit,
+                time_period= self.time_period,
+                bucket= self.bucket,
+                cooldown_id= self.cooldown_id,
+                check= self.check
+            )
+        
+        # Trigger Cooldown
+        self.trigger_cooldown = Cooldown(
+                limit= 1,
+                time_period= self.time_period,
+                bucket= self.bucket,
+                cooldown_id= self.cooldown_id,
+                check= self.check
+            )
+
+        if cooldown_id:
+            utils.shared_cooldown_refs[cooldown_id] = self.cooldown
```

Comment:
You'll need to figure out a way to nicely make this work for both. You define two cooldowns but only store 1 in the global store which may lead to issues

Reply:
The workaround that I have done is to store both cooldowns with the same ID only if the user defines an ID.

Remember that trigger cooldowns can not be shared. This could be added in the future because it will require an entirely new implementation of shared cooldowns only for trigger cooldowns.

Talking again about IDs:

```python
# Normal Cooldown
self.cooldown = Cooldown(
        limit= self.limit,
        time_period= self.time_period,
        bucket= self.bucket,
        cooldown_id= self.cooldown_id, # <-- Here...
        check= self.check
    )

# Trigger Cooldown
self.trigger_cooldown = Cooldown(
        limit= 1,
        time_period= self.time_period,
        bucket= self.bucket,
        cooldown_id= self.cooldown_id, # <-- Here...
        check= self.check
```

This means that, if the user wants to clean up the cooldown, should be able to do it with the ID. It will clean both cooldowns, but for me, that makes sense because they are together. These two cooldowns will always work in the same function, if you want to clean up one cooldown is probably because you want to remove the cooldown from your function, regardless of if it is the normal cooldown or the trigger cooldown. 

In my opinion, it is not necessary to have separate ID for both cooldowns, as they work like twins. But, if separated IDs are wanted, then just add a second argument to the constructor:
```python
class TriggerCooldown:

    def __init__(
            self,
            limit: int,
            time_period: Union[float, datetime.timedelta],
            bucket: Optional[CooldownBucketProtocol] = None,
            *,
            normal_cooldown_id: Optional[Union[int, str]] = None, # <-- Here...
            trigger_cooldown_id: Optional[Union[int, str]] = None, # <-- Here...
            check: Optional[MaybeCoro] = default_check,
    ):
```
And add them to their respective cooldowns, but again, I do not see this as useful and it could lead to other unknown errors.

Reply:
I'll keep in mind it to make them shareable.